vowpalwabbit/lda_core.cc

```diff
@@ -1234,7 +1234,7 @@ LEARNER::base_learner *lda_setup(vw &all)
 
   ld.decay_levels.push_back(0.f);
 
-  LEARNER::learner<lda> &l = init_learner(&ld, ld.compute_coherence_metrics ? learn_with_metrics : learn, 1 << all.weights.stride_shift());
+  LEARNER::learner<lda> &l = init_learner(&ld, ld.compute_coherence_metrics ? learn_with_metrics : learn, 1 << all.weights.stride_shift(), prediction_type::scalars);
```

Comment:
pred_type inherits from the base learner: https://github.com/JohnLangford/vowpal_wabbit/blob/master/vowpalwabbit/learner.h#L260  
just like predictions inherit from the base learner. 

We could imagine enforcing pred_type specification everywhere, but this seems awkward because there are some reductions which do not touch the prediction.